Add support for Pulumi.Docker

mvsmal opened the pull request:
This adds support for Pulumi.Docker. It is pretty straightforward.
I tested the results on one of our projects, works fine.

**IMPORTANT**
I had to change the comparison of `ResourceType` and `subNamespaceOrName` because in the Docker schema a few types have camelCase subnamespaces for some reason. Without this fix the generated code was incomplete. Please let me know if you have concerns regarding this change.

Also I have a couple of questions.
1. I noticed a pretty unusual for F# code style. I mean pipe operator at the end of lines and spacing for match cases to align on the right after the arrow (looks Haskell-ish). In the change mentioned above it makes the lines pretty long. Should I keep this code style in the further PRs?
2. There are quite a lot of trailing spaces in the files. I have a config in Rider to trim spaces on save. For this PR I had to select the changes very carefully due to lots of space trimming in the file. Is it ok to commit such trimming?

This is first PR to this project, please let me know if I should improve smth or if I missed smth. I'd be happy to fix.

UnoSD commented:
thank you for the contribution @mvsmal 

1. I have adjusted my code style to match the standard now, however, at the time I created the library it was unusual, as you saw (I had my reasons, but I decided to uniform in the end). I'd welcome a PR that changes the style for the whole library, but I know it will take me quite some time to review, so I wouldn't prioritize it; for now, I would keep it consistent with the rest of the library and (hopefully) someday, I'll change it.

2. I would try and be consistent also here, but I won't reject the PR just for spaces, as long as the code style is consistent. What I would do, is to separate space changes from functional changes to make commits more readable in different PRs.

For the comparison, it's fine, we have to hack around the oddities of the schema, however, as I was forced to do with K8s, I would make the change specific for Docker and not general for all the providers as it may negatively impact others/build performance (maybe?). Could you please make this change? Thanks.

mvsmal commented:
Hi @UnoSD,
I've introduced the condition to make Docker an exception for case insensitivity. Please have a look. It would be great to finally have the Docker package. Thanks.

UnoSD commented on Pulumi.FSharp.Docker/Pulumi.FSharp.Docker.fsproj:
```diff
@@ -0,0 +1,61 @@
+        <PackageLicenseExpression>GPL-2.0-only</PackageLicenseExpression>
+        <Title>$(MSBuildProjectName)</Title>
+        <RepositoryUrl>https://github.com/UnoSD/Pulumi.FSharp.Extensions</RepositoryUrl>
+        <PackageTags>fsharp pulumi azure</PackageTags>
```
we need to remove the azure tag and add the docker

UnoSD commented on Pulumi.FSharp.Docker/Pulumi.FSharp.Docker.fsproj:
```diff
@@ -0,0 +1,61 @@
+            <Using Namespace="System.IO" />
+            <Code Type="Fragment" Language="cs"><![CDATA[
+                var number = new Random(DateTime.Now.Millisecond).Next();
+                File.WriteAllText("Myriad.fs", $"module private Azure\n\nmodule Force = let private nonce = {number}");
```
This will needs to be Docker, too

UnoSD commented on Pulumi.FSharp.Docker/Pulumi.FSharp.Docker.fsproj:
```diff
@@ -0,0 +1,61 @@
+        <!-- This is required only to force build of the plugin -->
+        <ProjectReference Include="../Pulumi.FSharp.Myriad\Pulumi.FSharp.Myriad.fsproj" PrivateAssets="All" />
+
+        <ProjectReference Include="..\Pulumi.FSharp.Core\Pulumi.FSharp.Core.fsproj" PrivateAssets="All" />
```
Is the dependency on Core needed?

UnoSD commented:
Looks good overall, just a couple of changes in the fsproj file and I can merge/publish. Thanks.

mvsmal commented:
@UnoSD I have fixed the concerns, also removed the dependency on Pulumi.FSharp.Core, although I noticed that other projects always reference it, not sure why exactly.
